libclc/CMakeLists.txt

```diff
@@ -378,6 +379,18 @@ foreach( t ${LIBCLC_TARGETS_TO_BUILD} )
                 # have to be explicity declared in the soruce.
                 list( APPEND flags -Xclang -fdeclare-spirv-builtins)
 
+                # OpenCL 3.0 extensions
+                list( APPEND flags -cl-std=CL3.0 "SHELL:-Xclang" )
+                string(CONCAT CL_3_0_EXTENSIONS
+                  "-cl-ext="
+                  "+cl_khr_fp16,"
```

Comment:
Would this not enable FP16 for all libclc targets and if so is that safe?

Reply:
Using an outdated version of https://github.com/intel/llvm/pull/8100